src/table/components/withSelections.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { useContextSelector, TableContext } from '../context';
+import { getSelectionStyle } from '../utils/styling-utils';
+import { getCellSelectionState, SelectionActions, TSelectionActions } from '../utils/selections-utils';
+import { hocProps, SelectionState } from '../../types';
+
+export default function withSelections(CellComponent: (props: hocProps) => JSX.Element) {
+  const HOC = (props: hocProps) => {
+    const { cell, styling, announce } = props;
+    const cellSelectionState = useContextSelector(TableContext, (value) =>
+      getCellSelectionState(cell, value.selectionState as SelectionState)
```

Comment:
value is possible to be {}, as TableContext initially is {}. {}.selectionState goes wrong. 
So I think value.?selectionState is better.

Reply:
Well according to the typing it can be {}, but don't think it will ever be in practice. Once root runs the first time the values are initialized. So I think it is fine to cast here, but not sure. The other alternative is to sanity check everytime we use something from the context

Reply:
that's true. I debug it and find that it never be {}. Then I suggest 
`export const TableContext = createContext<AppContextInterface>({} as AppContextInterface);`

Reply:
ah that actually works with the props on ContextValue not being optional!